architecture/2024-08-bicepdeployment-controller.md

```diff
@@ -0,0 +1,693 @@
+link to that issue instead.
+-->
+
+#### Jon can deploy cloud (Azure or AWS) resources defined in Bicep manifests using Kubernetes tools
+
+Jon is an infrastructure operator for an enterprise company. His team manages a production application that is deployed on Kubernetes and uses cloud resources. Jon wants to declare all dependencies for his application in Kubernetes manifests, including cloud resources. He installs Radius on his cluster and uses the rad CLI to generate a custom resource from his Bicep manifest. Jon applies the custom resource to his cluster, and Radius deploys the cloud resources defined in the Bicep manifest. If he wants to update or delete the cloud resources, he can do so by re-applying the updated custom resource or deleting the custom resource.
+
+#### Jon can deploy Radius resources defined in Bicep manifests using Kubernetes tools
+
+Now that he can see that Radius can deploy cloud resources defined in Bicep manifests, Jon wants to take advantage of Radius tooling, such as the App Graph, and fully "Rad-ify" his application. He writes a Bicep manifest that defines a Radius container that connects to the cloud resources, and uses the rad CLI to generate a custom resource from the Bicep manifest. Jon applies the custom resource to his cluster, and Radius deploys the Radius resources defined in the Bicep manifest. Now, Jon can take advantage of Radius tooling, such as the Radius Dashboard and App Graph, to manage his application.
+
+## User Experience
+<!--
+If the change impacts the user experience, provide expected interaction 
+flow we aim to achieve through this proposal.
+
+When users interact with Radius through the CLI, include sample 
+input commands and their corresponding output. Include a bicep/helm code 
+sample, if this proposal involves updates to that experience.
+-->
+
+### Radius Container
+
+**Sample Input:**
+<!--
+Provide a sample CLI command input and/or bicep/helm code.
+-->
+
+#### `app.bicep`
+```bicep
+extension radius
+
+param application string
+param tag string
+param port int
+
+resource container 'Applications.Core/containers@2023-10-01-preview' = {
+  name: 'container'
+  properties: {
+    application: application
+    container: {
+      image: 'ghcr.io/radius-project/samples/demo:${tag}'
+      ports: {
+        web: {
+          containerPort: port
+        }
+      }
+    }
+  }
+}
+```
+
+#### `app.bicepparam`
+```bicep
+using 'app.bicep'
+
+param application = ''
+param tag = ''
+param port = 3000
+```
+
+**Sample Output:**
+<!--
+Provide a sample output for the inputs provided above.
+-->
+```
+> rad bicep generate-kubernetes app.bicep --parameters @app.bicepparam --parameters tag=latest --parameters appnetworking.bicepparam --outfile app.yaml
+
+Generating BicepDeployment resource...
+BicepDeployment resource generated at app.yaml
+
+To apply the BicepDeployment resource onto your cluster, run:
+kubectl apply -f app.yaml
+```
+
+#### `app.yaml`
```

Comment:
let's align the technical scenarios with the user scenarios

Reply:
let's write/replace these scenarios with one that has an app/storage account connection